scrabble_cheater_test.py

```diff
@@ -0,0 +1,120 @@
+        with two words.
+        """
+        # log = logging.getLogger( "Test")
+        global scrabble_board
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+        scrabble_board[5][1] = 'M'
+        scrabble_board[5][3] = 'N'
+        scrabble_board[7][1] = 'M'
+        scrabble_board[7][3] = 'N'
+        #import ipdb; ipdb.set_trace()
+        location = fit_word("CATASTROPHE", "c2")['locations'][0]
+        self.assertEqual(location['score'], 28)
+        self.assertEqual(location['location'], ((4, 2), (14, 2)))
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+
+    def test_fit_word_on_empty_row(self):
+        """
+        Tests fit_word's performance on an empty row that will intersect
+        with two words.
+        """
+        global scrabble_board
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+        scrabble_board[5][1] = 'M'
+        scrabble_board[5][3] = 'N'
+        scrabble_board[7][1] = 'M'
+        scrabble_board[7][3] = 'N'
+        location = fit_word("CATASTROPHE", "c2")['locations'][0]
+        self.assertEqual(location['score'], 28)
+        self.assertEqual(location['location'], ((4, 2), (14, 2)))
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+
+    def test_fit_word_on_occupied_column(self):
+        """
+        Tests fit_word's performance on a non-empty column that intersects
+        with two words.
+        """
+        global scrabble_board
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+        scrabble_board[5][1] = 'M'
+        scrabble_board[5][2] = 'A'
+        scrabble_board[5][3] = 'N'
+        scrabble_board[7][1] = 'M'
+        scrabble_board[7][2] = 'A'
+        scrabble_board[7][3] = 'N'
+        location = fit_word("CATASTROPHE", "c2")['locations'][0]
+        self.assertEqual(location['score'], 18)
+        self.assertEqual(location['location'], ((4, 2), (14, 2)))
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+        scrabble_board[5][1] = 'M'
+        scrabble_board[5][3] = 'N'
+        scrabble_board[7][1] = 'M'
+        scrabble_board[7][2] = 'A'
+        scrabble_board[7][3] = 'N'
+        location = fit_word("CATASTROPHE", "c2")['locations'][0]
+        self.assertEqual(location['score'], 23)
+        self.assertEqual(location['location'], ((4, 2), (14, 2)))
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+
+    def test_fit_word_on_occupied_row(self):
+        """
+        Tests fit_word's performance on a non-empty row that intersects
+        with two words.
+        """
+        global scrabble_board
+        scrabble_board = [[None for x in range(BOARD_SIZE)] for y in range(BOARD_SIZE)]
+        scrabble_board[1][5] = 'M'
```

Comment:
This kind of cumbersome manual setting almost begs for a nicer interface:

scrabble_board.add_horizontal_word('MAN', (1, 5))